### DISCUSSION TITLE
I am able to extract text from PDF but not its preserving the layout

### DISCUSSION THREAD

**Discussion by surendramudragada** [jsvine/pdfplumber#1165] [Ask for help with specific PDFs]
I have a pdf , using extract_text i am able to extract the text from the document.
But i need the text to maintain the same layout as in the source pdf document.
I tried to specify layout=true but it didnt work. 
Can you please help on this ? Will be truly grateful.

**Comment by jsvine** [comment]
Hi @surendramudragada, and thanks for your interest in `pdfplumber`.  It will be difficult to assist you, unfortunately, without a more detailed description and/or the PDF and output.

**Comment by Raj199365** [comment]
Hi, Is there any command to preserve tabs, spacings while extracting from searchable pdf file and writing it as formatted text file.
[Extracted using python script using pdfplumber.docx](https://github.com/user-attachments/files/16488576/Extracted.using.python.script.using.pdfplumber.docx)

**Reply by mkl-public** [reply]
> preserve tabs, spacings while extracting from searchable pdf file 

There are no "tabs" in PDFs. 

And the spacing between glyphs in a PDF may well not correspond to an integer multiple of the width of a space character. Also on a page different fonts with different space widths may be used while you display the extracted text using a single font with its own space width which is likely to create additional shifts. 

Nonetheless there are tools that try to preserve the layout somewhat but don't expect too much from them. Usually they have an option like `--layout` or "layout" in their name.
